// nextra-theme-blog/dist/cusdis.js
import {
  useBlogContext
} from "./chunk-BUW6DVV6.js";

// src/cusdis.tsx
import { useTheme } from "next-themes";
import dynamic from "next/dynamic";
import { useRouter } from "next/router";
import React from "react";
var Cusdis = dynamic(() => import("react-cusdis").then((mod) => mod.ReactCusdis), { ssr: false });
var Comments = () => {
  var _a;
  const { config, opts } = useBlogContext();
  const router = useRouter();
  const { theme, resolvedTheme } = useTheme();
  if (config.cusdis && !config.cusdis.appId) {
    console.warn("[nextra/cusdis] `appId` is required");
  }
  return ((_a = config.cusdis) == null ? void 0 : _a.appId) ? /* @__PURE__ */ React.createElement(Cusdis, {
    lang: config.cusdis.lang,
    style: { marginTop: "4rem" },
    attrs: {
      host: config.cusdis.host || "https://cusdis.com",
      appId: config.cusdis.appId,
      pageId: router.pathname,
      pageTitle: opts.title,
      theme: theme === "dark" || resolvedTheme === "dark" ? "dark" : "light"
    }
  }) : null;
};
var cusdis_default = Comments;
export {
  cusdis_default as default
};
